fix hardcoding of compiler commands/flags, use $CXX rather than $CC
author: Kenneth Hoste (HPC-UGent)
--- ea-utils.1.1.2-686/Makefile.orig	2014-04-07 19:26:35.000000000 +0200
+++ ea-utils.1.1.2-686/Makefile	2014-08-06 15:20:47.803786753 +0200
@@ -1,13 +1,13 @@
 #
 # $Id: Makefile 670 2013-12-13 16:47:07Z earonesty $
 
-CC=g++
+CXX=g++
 PREFIX?=/usr
 BINDIR?=$(PREFIX)/bin
-CFLAGS?=-O3 -I. 
+CXXFLAGS?=-O3 -I. 
 CPPFLAGS?=-O3 -I.
 # for debugging:
-# CFLAGS?=-g -I. 
+# CXXFLAGS?=-g -I. 
 # CPPFLAGS?=-g -I.
 
 PKG=ea-utils
@@ -64,24 +64,24 @@
 	cp $< $@
 
 %: %.c fastq-lib.cpp fastq-lib.h 
-	$(CC) $(CFLAGS) fastq-lib.cpp -o $@ $<
+	$(CXX) $(CXXFLAGS) fastq-lib.cpp -o $@ $<
 
 %: %.cpp fastq-lib.cpp fastq-lib.h
-	$(CC) $(CFLAGS) fastq-lib.cpp -o $@ $<
+	$(CXX) $(CXXFLAGS) fastq-lib.cpp -o $@ $<
 
 
 %: %.c gcModel.c gcModel.h
-	$(CC) $(CFLAGS) gcModel.c -o $@ $<
+	$(CXX) $(CXXFLAGS) gcModel.c -o $@ $<
 
 %: %.cpp gcModel.c gcModel.h
-	$(CC) $(CFLAGS) gcModel.c -o $@ $<
+	$(CXX) $(CXXFLAGS) gcModel.c -o $@ $<
 
 # why the libbam.a doesn't work?  not sure... *.o works
 sam-stats: sam-stats.cpp samtools/libbam.a samtools/bam.h fastq-lib.h
 ifeq ($(OS),Windows_NT)
-	$(CC) $(CFLAGS) samtools/*.o -lz -lpthread -lws2_32 fastq-lib.cpp $< -o $@
+	$(CXX) $(CXXFLAGS) samtools/*.o -lz -lpthread -lws2_32 fastq-lib.cpp $< -o $@
 else
-	$(CC) $(CFLAGS) samtools/*.o -lz -lpthread fastq-lib.cpp $< -o $@
+	$(CXX) $(CXXFLAGS) samtools/*.o -lz -lpthread fastq-lib.cpp $< -o $@
 endif
 
 samtools/libbam.a: samtools/*.c samtools/*.h
@@ -91,14 +91,14 @@
 ifeq ($(OS),Windows_NT)
 	echo varcall: not supported yet
 else
-	$(CC) $(CFLAGS) fastq-lib.cpp tidx/tidx-lib.cpp -o $@ $< -lgsl -lgslcblas
+	$(CXX) $(CXXFLAGS) fastq-lib.cpp tidx/tidx-lib.cpp -o $@ $< -lgsl -lgslcblas
 endif
 
 fastq-stats: fastq-stats.cpp fastq-lib.cpp gcModel.c
-	$(CC) $(CFLAGS) fastq-lib.cpp gcModel.c -o $@ $<
+	$(CXX) $(CXXFLAGS) fastq-lib.cpp gcModel.c -o $@ $<
 
 bam-filter:  bam-filter.cpp 
-	$(CC) $(CFLAGS) fastq-lib.cpp -o $@  $< -lbamtools 
+	$(CXX) $(CXXFLAGS) fastq-lib.cpp -o $@  $< -lbamtools 
 
 clean:
 	rm -f *.o $(BIN)
